servicetalk-log4j2-mdc/src/test/java/io/servicetalk/log4j2/mdc/ServiceTalkThreadContextMapLog4jProviderTest.java

```diff
@@ -15,18 +15,83 @@
  */
 package io.servicetalk.log4j2.mdc;
 
-import io.servicetalk.log4j2.mdc.utils.ServiceTalkThreadContextMap;
+import io.servicetalk.concurrent.api.Executors;
+import io.servicetalk.concurrent.api.Single;
 
 import org.apache.logging.log4j.ThreadContext;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.slf4j.MDC;
+
+import java.util.Map;
 
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.sameInstance;
 
 class ServiceTalkThreadContextMapLog4jProviderTest {
+
+    private static final String FOO_STRING = "foo";
+    private static final String BAR_STRING = "bar";
     @Test
     void testProviderLoadsClass() {
-        assertThat(ThreadContext.getThreadContextMap(), is(instanceOf(ServiceTalkThreadContextMap.class)));
+        assertThat(ThreadContext.getThreadContextMap(), is(instanceOf(DefaultServiceTalkThreadContextMap.class)));
+    }
+
+    @BeforeEach
+    void setup() {
+        MDC.clear();
```

Comment:
should this also be at least cleared after all so that it doesn't interfere with other tests that potentially depend on a clean MDC?

Reply:
I don't think it's strictly necessary now, but you're right in that it could pose issues in the future.